pkg/reconciler/v1alpha1/clusteringress/resources/virtual_service.go

```diff
@@ -120,6 +122,38 @@ func makeVirtualServiceRoute(hosts []string, http *v1alpha1.HTTPClusterIngressPa
 	}
 }
 
+func dedup(hosts []string) []string {
+	set := sets.NewString()
```

Comment:
This whole thing can be:
```
return sets.NewString(hosts...).List()  // This is sorted.
```

Reply:
I didn't do this from the get go, since I wasn't sure we want it sorted or "deterministic" (I think some unit tests failed)